File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/chat-openai-tool-call.js
'use strict'

const { finish } = require('../common')
const helpers = require('./helpers')

async function main () {
  const tracer = helpers.loadLangChainTracer()
  const model = helpers.chatOpenAI({ temperature: 0 }).bindTools([{
    type: 'function',
    function: {
      name: 'get_weather',
      description: 'Get the weather for a city.',
      parameters: { type: 'object', properties: { city: { type: 'string' } }, required: ['city'] },
    },
  }])
  await model.invoke([['user', 'What is the weather in Paris?']])
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack}\n`)
  process.exitCode = 1
})

File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/chat-openai-tool-call.py
from common import finish

from parity_langchain import chat_anthropic, chat_openai, openai_embeddings, start  # noqa: F401


start()
model = chat_openai(temperature=0).bind_tools(
    [
        {
            "type": "function",
            "function": {
                "name": "get_weather",
                "description": "Get the weather for a city.",
                "parameters": {"type": "object", "properties": {"city": {"type": "string"}}, "required": ["city"]},
            },
        }
    ]
)
model.invoke([("user", "What is the weather in Paris?")])
finish()
